Add --pretty-printed command to package-collection-generate

friedbunny opened the pull request:
Adds a `--pretty-printed` flag to package-collection-generate that nicely formats the collection output.

Fixes rdar://73019498.

friedbunny commented:
Hrm, looks like this is going to need a little bit of work to incorporate changes from #20 (which is obviously more important and should come first).

yim-lee commented on Sources/PackageCollectionGenerator/PackageCollectionGenerate.swift:
```diff
@@ -45,6 +45,9 @@ public struct PackageCollectionGenerate: ParsableCommand {
     private var revision: Int?
 
+    @Flag(name: .long, inversion: .prefixedNo, help: "Format output using friendly indentation and line-breaks.")
+    private var prettyPrinted: Bool = true
```
I think the default should be false since pretty-printing adds redundant new lines and whitespaces to the generated JSON document, which means larger download size. Package collections are primarily for machines, and humans can always use tools such as [jq](https://stedolan.github.io/jq/) to make them more readable.

friedbunny replied:
Works for me — changed the default back to non-pretty in b0121d9.

friedbunny commented:
Finally came back to this.

- Rebased and fixed conflicts.
- Default has changed back to non-pretty and the `--no-pretty-print` flag has been removed.

yim-lee reviewed (approved):
Thanks @friedbunny.

Can you please rebase again (sorry!) and also run `scripts/soundness.sh`?

friedbunny commented:
Rebased, fixed a couple conflicts, and ran `scripts/soundness.sh`. There were a few swiftformat suggestions in unrelated files that I ignored because they changed access levels, which seemed potentially suspect.

<details>
    <summary>Ignored swiftformat diff.</summary>

```diff
diff --git a/Sources/PackageCollectionGenerator/Models/PackageCollectionGeneratorInput.swift b/Sources/PackageCollectionGenerator/Models/PackageCollectionGeneratorInput.swift
index 125bd4e..e7c9b1d 100644
--- a/Sources/PackageCollectionGenerator/Models/PackageCollectionGeneratorInput.swift
+++ b/Sources/PackageCollectionGenerator/Models/PackageCollectionGeneratorInput.swift
@@ -62,9 +62,9 @@ extension PackageCollectionGeneratorInput: CustomStringConvertible {
     }
 }
 
-extension PackageCollectionGeneratorInput {
+public extension PackageCollectionGeneratorInput {
     /// Represents a package to be processed
-    public struct Package: Equatable, Codable {
+    struct Package: Equatable, Codable {
         /// The URL of the package. Currently only Git repository URLs are supported.
         public let url: URL
 
diff --git a/Sources/PackageCollectionGenerator/PackageMetadataProviders/GitHubPackageMetadataProvider.swift b/Sources/PackageCollectionGenerator/PackageMetadataProviders/GitHubPackageMetadataProvider.swift
index 8484cfc..83326bc 100644
--- a/Sources/PackageCollectionGenerator/PackageMetadataProviders/GitHubPackageMetadataProvider.swift
+++ b/Sources/PackageCollectionGenerator/PackageMetadataProviders/GitHubPackageMetadataProvider.swift
@@ -155,8 +155,8 @@ struct GitHubPackageMetadataProvider: PackageMetadataProvider {
     }
 }
 
-extension GitHubPackageMetadataProvider {
-    fileprivate struct GetRepositoryResponse: Codable {
+private extension GitHubPackageMetadataProvider {
+    struct GetRepositoryResponse: Codable {
         let name: String
         let fullName: String
         let description: String?
@@ -171,8 +171,8 @@ extension GitHubPackageMetadataProvider {
     }
 }
 
-extension GitHubPackageMetadataProvider {
-    fileprivate struct Readme: Codable {
+private extension GitHubPackageMetadataProvider {
+    struct Readme: Codable {
         let url: Foundation.URL
         let htmlURL: Foundation.URL
         let downloadURL: Foundation.URL
@@ -184,7 +184,7 @@ extension GitHubPackageMetadataProvider {
         }
     }
 
-    fileprivate struct License: Codable {
+    struct License: Codable {
         let url: Foundation.URL
         let htmlURL: Foundation.URL
         let downloadURL: Foundation.URL
diff --git a/Sources/Utilities/Git.swift b/Sources/Utilities/Git.swift
index 037ed1a..c714239 100644
--- a/Sources/Utilities/Git.swift
+++ b/Sources/Utilities/Git.swift
@@ -78,8 +78,9 @@ public struct GitURL {
             let regex = try NSRegularExpression(pattern: #"([^/@]+)[:/]([^:/]+)/([^/.]+)(\.git)?$"#, options: .caseInsensitive)
             if let match = regex.firstMatch(in: gitURL, options: [], range: NSRange(location: 0, length: gitURL.count)) {
                 if let hostRange = Range(match.range(at: 1), in: gitURL),
-                    let ownerRange = Range(match.range(at: 2), in: gitURL),
-                    let repoRange = Range(match.range(at: 3), in: gitURL) {
+                   let ownerRange = Range(match.range(at: 2), in: gitURL),
+                   let repoRange = Range(match.range(at: 3), in: gitURL)
+                {
                     let host = String(gitURL[hostRange])
                     let owner = String(gitURL[ownerRange])
                     let repository = String(gitURL[repoRange])
diff --git a/Tests/PackageCollectionSignerExecutableTests/PackageCollectionSignTests.swift b/Tests/PackageCollectionSignerExecutableTests/PackageCollectionSignTests.swift
index 69b805f..0bc8382 100644
--- a/Tests/PackageCollectionSignerExecutableTests/PackageCollectionSignTests.swift
+++ b/Tests/PackageCollectionSignerExecutableTests/PackageCollectionSignTests.swift
@@ -64,7 +64,8 @@ private struct MockPackageCollectionSigner: PackageCollectionSigner {
               certChainPaths: [URL],
               certPrivateKeyPath: URL,
               certPolicyKey: CertificatePolicyKey,
-              callback: @escaping (Result<Model.SignedCollection, Error>) -> Void) {
+              callback: @escaping (Result<Model.SignedCollection, Error>) -> Void)
+    {
         let signature = Model.Signature(
             signature: "test signature",
             certificate: Model.Signature.Certificate(
```

</details>
